glib-macros/src/static_attribute.rs

```diff
@@ -0,0 +1,43 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use proc_macro::TokenStream;
+use proc_macro_error::abort_call_site;
+use quote::quote;
+use syn::ItemFn;
+
+pub fn impl_static(input: ItemFn) -> TokenStream {
+    let vis = &input.vis;
+    let block = &input.block;
+    let name = &input.sig.ident;
+    let output = &input.sig.output;
+    
+    // We need to get the inner type T out of &'static [T]
+    let output_type = match output {
+        syn::ReturnType::Type(_, ret_type) => match &**ret_type {
```

Comment:
My question here, is whether we would write something like `fn signals() -> &'static [Signal]`, if that's the case, we would need a way to get `T` from the return type so that I can use it in `Vec<T>` below.

Reply:
In other words, would the the macro be used only for those two use cases for now: properties/signals?

Reply:
I think you could implement it a bit more generically. The user would write `fn signals() -> &'static [Signal]`, you extract the type: `[Signal]`. It's a slice, so you use a `Vec<_>` internally.

Similarly if you the is some plain `T`, you would work with that directly. If it's a `str` you would work with `String`, if it's a `Path` you would work with `PathBuf`. And that should cover most interesting cases for now.

Reply:
> I think you could implement it a bit more generically. The user would write fn signals() -> &'static [Signal], you extract the type: [Signal]. It's a slice, so you use a Vec<_> internally.

Right, that's what it does now

> Similarly if you the is some plain T, you would work with that directly. If it's a str you would work with String, if it's a Path you would work with PathBuf. And that should cover most interesting cases for now.

After a bit of research,  I couldn't find any nice way to special case `Path`. I pushed something but it's very hackish. I was suggested to use traits for such situation  but I'm not sure whether it's worth it here.

Reply:
Well there's [`ToOwned`](https://doc.rust-lang.org/std/borrow/trait.ToOwned.html), maybe that can actually be used here. It covers the `Vec`, `String` and `PathBuf` cases at least, and has a blanket implementation for everything that is `Clone`.